hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/EntitySelectFetchInitializer.java

```diff
@@ -193,6 +199,11 @@ public void initializeInstance(RowProcessingState rowProcessingState) {
 			if ( toOneMapping.getNotFoundAction() == NotFoundAction.EXCEPTION ) {
 				throw new FetchNotFoundException( entityName, entityIdentifier );
 			}
+			rowProcessingState.getJdbcValuesSourceProcessingState()
```

Comment:
In `AbstractEntityInitializer` we always register the loading entity entry on resolve, so IMO we should do this for select fetching as well.

Reply:
with `internaLoad()` if the entity is not null we have it in the PC, so no need to register it, and before calling the `internalLoad()` we have no entityInstance to register.

Reply:
Ok, you are right. We always check the PC first before looking for a loading entity entry, so this is fine.